torch_xla/csrc/ops/index_get.cpp

```diff
@@ -0,0 +1,50 @@
+#include "torch_xla/csrc/ops/index_get.h"
+
+#include "tensorflow/compiler/xla/xla_client/debug_macros.h"
+#include "tensorflow/compiler/xla/xla_client/util.h"
+#include "torch_xla/csrc/lowering_context.h"
+#include "torch_xla/csrc/ops/infer_output_shape.h"
+#include "torch_xla/csrc/xla_lower_util.h"
+
+namespace torch_xla {
+namespace ir {
+namespace ops {
+namespace {
+
+xla::Shape NodeOutputShape(const Value& base, const Value& indices,
+                           xla::int64 start_dim) {
+  auto lower_for_shape_fn =
+      [start_dim](tensorflow::gtl::ArraySlice<const xla::XlaOp> operands)
+      -> xla::XlaOp {
+    XLA_CHECK_EQ(operands.size(), 2);
+    return CreateIndex(operands[0], operands[1], start_dim);
+  };
+  return InferOutputShape({base.shape(), indices.shape()}, lower_for_shape_fn);
+}
+
+}  // namespace
+
+IndexGet::IndexGet(const ir::Value& base, const ir::Value& indices,
+                   xla::int64 start_dim)
+    : Node(
+          OpKind(at::aten::index), {base, indices},
```

Comment:
Is this really `aten::index` or should be `xla::index_get`?

Reply:
It's `aten::index`, the other indexing operations have dedicated names (`index_select`, `index_put` etc). I like `IndexGet` better as a name, but we do have an ATen symbol.